components/date-picker/internal/DatePickerDatetimePanel.razor.cs

```diff
@@ -195,26 +194,30 @@ private async Task ScrollToSelectedTimeIfChangedAsync()
 
             if (currentValue.Hour != _selectedHour)
             {
-                await ScrollToSelectedHourAsync();
+                await ScrollToSelectedHourAsync(currentValue);
             }
             if (currentValue.Minute != _selectedMinute)
             {
-                await ScrollToSelectedMinuteAsync();
+                await ScrollToSelectedMinuteAsync(currentValue);
             }
             if (currentValue.Second != _selectedSecond)
             {
-                await ScrollToSelectedSecondAsync();
+                await ScrollToSelectedSecondAsync(currentValue);
             }
         }
 
         private void DatePicker_OverlayVisibleChanged(object sender, bool visible)
         {
-            if (visible)
+            if (!visible || Value is null)
             {
-                _ = ScrollToSelectedHourAsync(0);
-                _ = ScrollToSelectedMinuteAsync(0);
-                _ = ScrollToSelectedSecondAsync(0);
+                return;
             }
+
+            var currentValue = Value.Value;
+
+            _ = ScrollToSelectedHourAsync(currentValue, 0);
+            _ = ScrollToSelectedMinuteAsync(currentValue, 0);
+            _ = ScrollToSelectedSecondAsync(currentValue, 0);
```

Comment:
I think this can be combined into one interop

Reply:
> I think this can be combined into one interop

Do you mean javascript interop?

Reply:
That's right.

Reply:
I'm unsure how to combine it in one javascript call and if it is worth it. This will require adding more javascript. From my point of view, the input panel re-renders itself way too often, which should be looked into. Also, maybe there is a way to integrate scroll in the time-panel-column rendering process 🤔

Reply:
Yes, we need to minimize the number of renderings.